diff --git a/crates/fub-features/tests/undo.rs b/crates/fub-features/tests/undo.rs
--- a/crates/fub-features/tests/undo.rs
+++ b/crates/fub-features/tests/undo.rs
@@ -611,6 +611,7 @@ impl VaultStorage for SupportThatExplodes {
         path: &camino::Utf8Path,
         merge: fub_kernel::storage::Merge<'_>,
     ) -> std::io::Result<()> {
+        self.maybe_explodes(path);
         self.inner.update(path, merge)
     }
     fn append(&self, path: &camino::Utf8Path, bytes: &[u8]) -> std::io::Result<()> {
